packages/material-ui/test/typescript/moduleAugmentation/buttonCustomProps.spec.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+import { createMuiTheme } from '@material-ui/core/styles';
+
+// Update the Button's extendable props options
+declare module '@material-ui/core/Button' {
+  interface ButtonPropsVariantOverrides {
+    dashed: true;
+  }
+  interface ButtonPropsColorOverrides {
+    success: true;
+  }
+  interface ButtonPropsSizeOverrides {
+    extraLarge: true;
+  }
+}
+
+// theme typings should work as expected
+const theme = createMuiTheme({
+  components: {
+    MuiButton: {
+      variants: [
+        {
+          props: { variant: 'dashed' },
+          style: {
+            border: `2px dashed grey`,
+          },
+        },
+        {
+          props: { size: 'extraLarge' },
+          style: {
+            '& > *:nth-of-type(1)': {
+              fontSize: 26,
+            },
```

Comment:
I would hope it's enough 👼 

```suggestion
            fontSize: 26,
```

Reply:
I just copy this from the Button's styles, but it's not even that important here. Let's simplify :)